docs/design/storage-node-deployment-windows-home.md

```diff
@@ -0,0 +1,146 @@
+# Storage Node Automatic Updates and Installation for Windows Home
+
+## Overview
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date.
+If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.
+
+Currently we are using Docker for automatic updates which means we require Docker to be installed.
+Docker is being used for many other things as well, so we need to cover these cases as well.
+
+## Goals
+
+Docker is being used for:
+
+* Automatic Updates (with watchtower)
+* Restarting on crash
+* Logging (kind of)
+
+To deploy the automatic updates we need to handle these cases.
+
+Docker is not supported on windows home so we will ensure an automatic update system is built into the Storage nodes. We also need to ensure that it plays nicely with all the common anti-viruses and firewalls.
+
+## Services
+* Installer (msi)
+    * Installs automatic updater binary and error gui application
+    * Sets up automatic updater Service with sufficient privileges.
+* Automatic Updater (binary)
+    * Downloads, Sets up, and runs the storage node.
+    * If storage node has not been setup we don't want to try to run the storage node.
+    * Send error reports to satellite.
+    * Writes errors to log file
+* Storage Node (binary)
+    * shares drive with satellite network.
+* Error gui application
+    * shows errors from log file
+    * notifies user of service errors.
+    * saves last reported error timestamp to a file for knowing if there are unread errors.
+
+### Automatic Updates
+
+Finding out the minimum version and latest stable.
+
+General
+* Windows firewall can block storage node operations.
+    * [isportallowed](https://docs.microsoft.com/en-us/windows/win32/api/netfw/nf-netfw-inetfwmgr-isportallowed) windows api function can be used to make sure we are allowed through firewall
+    * Can we add code to detect if firewall is blocking storage node operations?
+    * Unblock storage node operator in firewall settings.
+    * Can we detect if windows firewall is running?
+
+Download a new binary.
+* download fails
+    * Log error in log file if possible
```

Comment:
For goals section we don't need to specify the action, just describe the problem.